### PR TITLE
 [WIP] Test cases changed to use custom storageclass names [tier1, tier2]

### PR DESCRIPTION
**Pull request by paraggit**
[WIP] Test cases changed to use custom storageclass names

### CODE REVIEW

File: ocs_ci/helpers/helpers.py [hunk 1]
```diff
@@ -632,41 +632,17 @@ def default_storage_class(
...
+    if interface_type in [
+        constants.CEPHBLOCKPOOL,
+        constants.OCS_COMPONENTS_MAP["blockpools"],
+    ]:
+        resource_name = storageclass_name(constants.OCS_COMPONENTS_MAP["blockpools"])
+    elif interface_type in [
+        constants.CEPHFILESYSTEM,
+        constants.OCS_COMPONENTS_MAP["cephfs"],
+    ]:
+        resource_name = storageclass_name(constants.OCS_COMPONENTS_MAP["cephfs"])
```
Comment (ebenahar) [inline ocs_ci/helpers/helpers.py:645]: does it cover also external mode SCs?

File: ocs_ci/helpers/storageclass_helpers.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+
+
+def storageclass_name(interface, external_mode=False):
```
Comment (ebenahar) [inline ocs_ci/helpers/storageclass_helpers.py:5]: suggest renaming to `get_default_storage_class_name`

File: tests/manage/add_metadata_feature/test_metadata.py [hunk 3]
```diff
@@ -61,9 +62,11 @@ def test_metadata_feature_unavailable_for_previous_versions(
...
         for csi-cephfsplugin-provisioner and csi-rbdplugin-provisioner pods
         """
-        external_mode = config.DEPLOYMENT["external_mode"]
-        fs, sc_name = metadata_utils.update_testdata_for_external_modes(
-            sc_name, fs, external_mode=external_mode
+        sc_name = storageclass_name(sc_interface)
+        fs = (
+            ""
+            if sc_interface == constants.OCS_COMPONENTS_MAP["blockpools"]
+            else "fsvol001"
```
Comment (ebenahar) [inline tests/manage/add_metadata_feature/test_metadata.py:69]: please save "fsvol001" as a constant

Comment (paraggit) [reply tests/manage/add_metadata_feature/test_metadata.py:69]: "fsvol001" is just any random string not a specific format value , Hence it has uses like that

File: ocs_ci/helpers/storageclass_helpers.py (hunk 2 above)
Comment (paraggit) [reply ocs_ci/helpers/storageclass_helpers.py]: Done

File: ocs_ci/helpers/helpers.py (hunk 1 above)
Comment (paraggit) [reply ocs_ci/helpers/helpers.py]: Yes , It will work for external mode too.

File: ocs_ci/helpers/storageclass_helpers.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    """
+    This Function will return the default storage class name from the cluster.
+    This will return only "cephfs", "rbd" and "rgw" type storage class names only.
+    Args:
+        interface (str): storage class interface type
+        external_mode (bool) : True if external mode setup else False.
+    Returns:
+        (str): storage class name
+    """
+
```
Comment (ebenahar) [inline ocs_ci/helpers/storageclass_helpers.py:15]: please add an info log message `getting the default storageclass name`

Comment (paraggit) [reply ocs_ci/helpers/storageclass_helpers.py:19]: Added log message

**Review by ebenahar — APPROVED** [review]

File: tests/manage/mcg/ui/test_mcg_ui.py [hunk 5]
```diff
@@ -331,11 +332,11 @@ def test_obc_creation_and_deletion(
...
         obc_ui_obj = ObjectBucketClaimsTab()
 
-        if (
-            config.DEPLOYMENT["external_mode"]
-            and storageclass == "ocs-storagecluster-ceph-rgw"
-        ):
-            storageclass = "ocs-external-storagecluster-ceph-rgw"
+        if storageclass == constants.DEFAULT_STORAGECLASS_RGW:
+            storageclass = get_default_storage_class_name(
+                constants.OCS_COMPONENTS_MAP["rgw"]
```
Comment (DanielOsypenko) [inline tests/manage/mcg/ui/test_mcg_ui.py:337]: should be `get_default_storage_class_name(constants.OCS_COMPONENTS_MAP["rgw"], external_mode=True)`

**Review by avd-sagare — APPROVED** [review]

**Pull request closed without merging**